consensus/reactor.go

```diff
@@ -639,7 +639,15 @@ OUTER_LOOP:
 			// Load the block commit for prs.Height,
 			// which contains precommit signatures for prs.Height.
 			commit := conR.conS.blockStore.LoadBlockCommit(prs.Height)
-			if ps.PickSendVote(commit) {
+			valSet, err := conR.conS.blockExec.LoadValidators(prs.Height)
+			if err != nil {
+				// This should never happen as we should have the validator
+				// set for this height by now. In any case, don't panic.
```

Comment:
what would be the consequences of that (not panicking)?

Reply:
We might keep trying to send the peer a commit from a height where we don't have the validator set until the peer moves on (perhaps by receiving the commit votes from someone else). 

Though maybe we should sleep here before continue to avoid a hot loop.

Reply:
So I'm not sure exactly what to do here anymore. Maybe we should panic. Maybe we can eliminate ever even having an error here. Don't think it needs to block the PR. See https://github.com/tendermint/tendermint/issues/3596